proxy/proxy/connect_tls_117.go

```diff
@@ -38,7 +39,8 @@ func (c *Client) connectTLS(
 	// this file is conditionally compiled on only Go versions >= 1.17.
 	if err := ret.HandshakeContext(ctx); err != nil {
 		_ = ret.Close()
-		c.invalidateCfg(cfg, instance)
+		err = fmt.Errorf("config invalidated after TLS handshake failed, error = %w", err)
```

Comment:
nit: consider moving the wrapping to add context ("config invalidated because error:") inside the invalidate function

Reply:
Good idea. Should have thought of that myself.